beacon-chain/rpc/eth/node/handlers.go

```diff
@@ -0,0 +1,43 @@
+package node
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+// GetSyncStatus requests the beacon node to describe if it's currently syncing or not, and
+// if it is, what block it is up to.
+func (s *Server) GetSyncStatus(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "node.GetSyncStatus")
+	defer span.End()
+
+	start := time.Now()
+	isOptimistic, err := s.OptimisticModeFetcher.IsOptimistic(ctx)
+	if err != nil {
+		http2.HandleError(w, "Could not check optimistic status: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	log.Warnf("IsOptimistic: %s", time.Since(start))
+
+	headSlot := s.HeadFetcher.HeadSlot()
```

Comment:
hmm is the headslot fetcher always supposed to be before the syncing and execution client checks?

is this something we should group together for safer calls in the future?